src/character_attire.cpp

```diff
@@ -2230,6 +2230,61 @@ void outfit::pickup_stash( const item &newit, int &remaining_charges, bool ignor
     }
 }
 
+static std::vector<pocket_data_with_parent> get_child_pocket_with_parent(
+    const item_pocket *pocket, const item_location &parent, item_location it, const int nested_level )
+{
+    std::vector<pocket_data_with_parent> ret;
+    if( pocket != nullptr ) {
+        pocket_data_with_parent pocket_data = { pocket, item_location::nowhere, nested_level };
+        const item_location new_parent = item_location( it );
+
+        if( parent != item_location::nowhere ) {
+            pocket_data.parent = item_location( parent, it.get_item() );
+        }
+
+        for( const item *contained : pocket->all_items_top() ) {
+            const item_location poc_loc = item_location( it, const_cast<item *>( contained ) );
+            for( const item_pocket *pocket_nest : contained->get_all_contained_pockets() ) {
+                std::vector<pocket_data_with_parent> child =
+                    get_child_pocket_with_parent( pocket_nest, new_parent,
+                                                  poc_loc, nested_level + 1 );
+                for( const pocket_data_with_parent &pock_d : child ) {
+                    ret.emplace_back( pock_d );
+                }
+            }
+        }
+        ret.emplace_back( pocket_data );
+    }
+    return ret;
+}
+
+std::vector<pocket_data_with_parent> outfit::get_all_pocket_with_parent(
+    Character &guy )
+{
+    std::vector<pocket_data_with_parent> ret;
+
+    // Collect all pockets
+    std::list<item *> items;
+    item_location carried_item = guy.get_wielded_item();
+    if( carried_item != item_location::nowhere ) {
+        items.emplace_back( &*carried_item );
+    }
+    for( item &i : guy.worn.worn ) {
+        items.emplace_back( &i );
+    }
+    for( item *i : items ) {
+        for( const item_pocket *pocket : i->get_all_contained_pockets() ) {
+            item_location loc = item_location::nowhere;
+            std::vector<pocket_data_with_parent> child =
+                get_child_pocket_with_parent( pocket, loc, item_location( guy, i ), 0 );
+            for( const pocket_data_with_parent &pock_d : child ) {
+                ret.emplace_back( pock_d );
+            }
```

Comment:
```suggestion
            ret.insert( ret.end(), child.begin(), child.end() );
```

Reply:
These changes are committed in batches just before committing after changing other parts of the file ( to avoid consuming extra watts in CI ).